client/src/components/History/CurrentHistory/HistoryOperations.vue

```diff
@@ -209,12 +213,20 @@ export default {
             this.runOnSelection(purgeSelectedContent);
         },
         async runOnSelection(fn) {
-            const items = Array.from(this.contentSelection.values());
-            const type_ids = items.map((o) => o.type_id);
-            await fn(this.history, type_ids);
-            this.$emit("hide-selection", items);
+            const items = this.getExplicitlySelectedItems();
+            const filters = getFilters(this.filterText);
```

Comment:
We should probably use `getFilterDict` here instead, then there is no need for additional processing in the bulk query module and `getQueryDictFromFilters` can be removed again.